pkg/collector/data/examples_generator.go

```diff
@@ -276,25 +275,90 @@ type Rule struct {
 	Dest                 string
 	DestinationsExcluded bool
 	Services             []string
+	Conn                 *netset.TransportSet
 	Action               string
 	Direction            string // if not set, used as default with "IN_OUT"
 	Description          string
 }
 
-func (r *Rule) toNSXRule() *nsx.Rule {
-	return &nsx.Rule{
-		DisplayName:          &r.Name,
-		RuleId:               &r.ID,
-		Action:               (*nsx.RuleAction)(&r.Action),
-		SourceGroups:         []string{r.Source},
-		DestinationGroups:    []string{r.Dest},
-		SourcesExcluded:      r.SourcesExcluded,
-		DestinationsExcluded: r.DestinationsExcluded,
-		Services:             r.Services,
-		Direction:            r.directionStr(),
-		Scope:                []string{AnyStr}, // TODO: add scope as configurable
-		Description:          &r.Description,
+func (r *Rule) toCollectorRule() collector.Rule {
```

Comment:
why does the returned type need to be changed, if `nsx.Rule` also has `ServiceEntries` field?

Reply:
since go does not support abstract with json, the ServiceEntries in nsx.rule can not be used 
(nsx.rule.ServiceEntries is a slice of structs, while  collector..rule.ServiceEntries is a list of interfaces)